snapd.go

```diff
@@ -249,13 +249,11 @@ func action(ctx *cli.Context) {
 	setMaxProcs(cfg.GoMaxProcs)
 
 	c := control.New(cfg.Control)
-
-	coreModules = []coreModule{}
-
-	coreModules = append(coreModules, c)
 	s := scheduler.New(cfg.Scheduler)
+	c.SetTaskScheduler(s)
 	s.SetMetricManager(c)
-	coreModules = append(coreModules, s)
+
+	coreModules = []coreModule{c, s}
```

Comment:
> What do you think is the way to add 2-way communication to that interface?
I thought just about removing SetTaskScheduler() from control, modify scheduler's SetMetricManager() to something like:
```go
// Set metricManager for scheduler
func (s *scheduler) SetMetricManager(mm managesMetrics) {
	s.metricManager = mm
	mm.SetTaskScheduler(s) // <-- Just add this line
	schedulerLogger.WithFields(log.Fields{
		"_block": "set-metric-manager",
	}).Debug("metric manager linked")
}
```
And leave snapd.go unmodified, but in the end there is actually no difference as you said that plugin manager shouldn't know about scheduler.

Only possible way here would be to create something between scheduler and plugin manager, that could exchange communication between these modules (for example package with interface both for scheduler and plugin manager). But creating whole module just for exchanging data between two other modules makes a little mess, I think.

The best possible way in my opinion would be some sort of event manager shared between all modules (now every module has it's own event management), so it would be easy to pass events between modules. But on the other hand, this is really huge modification.

What do you think will be best way to do?